samples/sample_common/src/vaapi_allocator.cpp

```diff
@@ -503,6 +506,17 @@ mfxStatus vaapiFrameAllocator::LockFrame(mfxMemId mid, mfxFrameData *ptr)
                 }
                 else mfx_res = MFX_ERR_LOCK_MEMORY;
                 break;
+            case VA_FOURCC_ABGR:
+                if (mfx_fourcc == MFX_FOURCC_BGR4)
+                {
+                    ptr->Pitch = (mfxU16)vaapi_mid->m_image.pitches[0];
+                    ptr->R = pBuffer + vaapi_mid->m_image.offsets[0];
+                    ptr->G = pBuffer + vaapi_mid->m_image.offsets[1];
+                    ptr->B = pBuffer + vaapi_mid->m_image.offsets[2];
+                    ptr->A = ptr->R + 3;
```

Comment:
Wow. Interesting that VAImage supports only 3 planes as a maximum. This potentially can be a problem if someone will try to play with the offsets. @XinfengZhang: pay attention on this. A candidate for the VAAPI 3.x series:).

Reply:
ARGB is single plain, 4 samples format e.g. all the colors/Alpha channel are interleaved.
RGBP is 3 plain format where each color placed in separate plain. There are no format for 4 plains.

Reply:
I believe here you are mixing plains vs samples(colors)

Reply:
Yep, I can mix that up. Though only one of 2 ways could have been valid and programmed in the driver. @IlyaSergeyev : what iHD driver returns for:
- VAImage::num_planes ?
- VAImage::offsets[n], where n=0 .. num_planes-1 ?

Per Artem this should be VAImage::num_planes=1 with only 1st valid offsets. I tend to agree with Artem that this is what we should expect meaning that what you have written initially should be correct. However, I would like to understand driver behavior.